.github/workflows/test-repositories.yml

```diff
@@ -0,0 +1,91 @@
+# This workflow is provided via the organization template repository
+#
+# https://github.com/nextcloud/.github
+# https://docs.github.com/en/actions/learn-github-actions/sharing-workflows-with-your-organization
+
+name: Generate OpenAPI in Repositories
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+      - master
+      - stable*
+
+permissions:
+  contents: read
+
+concurrency:
+  group: openapi-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  openapi-repositories:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        php-versions: [ "8.1" ]
+        repositories: [ "nextcloud/server", "nextcloud/spreed" ]
+
+    name: openapi-repositories
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
+
+      - name: Set up php ${{ matrix.php-versions }}
+        uses: shivammathur/setup-php@4bd44f22a98a19e0950cbad5f31095157cc9621b # v2
+        with:
+          php-version: ${{ matrix.php-versions }}
+          coverage: none
+          ini-file: development
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Checkout repository for testing
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+        with:
+          repository: ${{ matrix.repositories }}
+          path: temp-repository
+
+      - name: Install dependencies
+        run: composer i
+
+      - name: Generate OpenAPI
+        if: matrix.repositories != 'nextcloud/server'
+        working-directory: temp-repository/
+        run: ../generate-spec
+
+      - name: Generate OpenAPI - Core
+        if: matrix.repositories == 'nextcloud/server'
+        working-directory: temp-repository/
+        run: ../generate-spec 'core' 'core/openapi.json'
+
+      - name: Generate OpenAPI - Server apps
+        if: matrix.repositories == 'nextcloud/server'
+        working-directory: temp-repository/
+        run: |
+          for path in apps/*/openapi.json; do
+            ../generate-spec "$(dirname "$path")" "$path" || exit 1
+          done
+
+      - name: Show changes of the API for assistance
+        working-directory: temp-repository/
+        run: |
+            git status
+            git --no-pager diff
+
+  summary:
+    permissions:
+      contents: none
+    runs-on: ubuntu-latest
+    needs: openapi-repositories
+
+    if: always()
+
+    name: openapi-repositories-summary
+
+    steps:
+      - name: Summary status
+        run: if ${{ needs.openapi-repositories.result != 'success' && needs.openapi-repositories.result != 'skipped' }}; then exit 1; fi
```

Comment:
Why do we need this? These jobs should not be set to required because sometimes there will be changes that cause a diff that was intended

Reply:
it's not failing on diff, only when `generate-specs` does a non-0 exit

Reply:
I thought the point was that it fails on diffs so that we can see when a PR changes the output

Reply:
It shows the diff and fails when the command fails, so the best of both worlds I would say 😅